### ISSUE TITLE
Advanced duplicate finder.

### ISSUE THREAD

**Issue by Worgle123** [files-community/Files#11385]
### What feature or improvement do you think would benefit Files?

I have pretty massive numbers of files, and I have been constantly frustrated about the default duplicate finder, which only appears to find duplicate names. I would love to see this feature find it's way to Files, as at the moment, I have to use paid clunky software, which doesn't even appear to do a great job. Something that had a look at file contents or even just size would be amazing!

### Requirements

Will add a advanced duplicate finder.

### Files Version

2.0.31.0

### Windows Version

Windows 11

### Comments

Could be both auto (when a file with the same name is added to a folder) and have an option to scan specific file areas.

**Comment by cinqmilleans** [comment]
This is a very useful feature that brings real added value. Unfortunately, having tried several, it seems too complex to me to integrate. It takes too many options to handle. You have to display millions of results and keep a user-friendly interface. Stains can take several hours. It is software in itself, but unfortunately there are no good free ones.

**Comment by Worgle123** [comment]
> This is a very useful feature that brings real added value. Unfortunately, having tried several, it seems too complex to me to integrate. It takes too many options to handle. You have to display millions of results and keep a user-friendly interface. Stains can take several hours. It is software in itself, but unfortunately there are no good free ones.

Interesting that it would be so complex. Maybe further down the road though? It could be improved over time, and slowly built up, with more feautures. Any improvement would really be very welcome here! Thanks!

**Comment by cinqmilleans** [comment]
Of course, but you have to be aware of the work to be done. You will need a powerful database. You shouldn't interfere with the rest of the application either (in terms of performance and UX). Maybe it would be better to make it a spinoff app.

**Comment by manfromarce** [comment]
It may be reasonable in my opinion to implement only manual scans via a context menu item that opens a separate window (such as Properties) and limit to one simultaneous scan. In this way it would be helpful but would't impact performance and the app in general too much as automatic background scans would do. It would also be less complex, e.g. I don't think CCleaner's duplicates finder mantains a database.

**Comment by Worgle123** [comment]
> It may be reasonable in my opinion to implement only manual scans via a context menu item that opens a separate window (such as Properties) and limit to one simultaneous scan. In this way it would be helpful but would't impact performance and the app in general too much as automatic background scans would do. It would also be less complex, e.g. I don't think CCleaner's duplicates finder mantains a database.

Just saw this now (6 March) I agree that 1 simultaneous scan would be a better proposal. How would you propose it would scan for duplicates? I thought that It could have a size filter, and it could have varying levels of sensitivity. Maybe it could even set a computer proposed sensitivity.

Thanks!

**Comment by Josh65-2201** [comment]
It could be possible to use a hash value to check. but that would only find exact duplicates in file data.

**Comment by manfromarce** [comment]
It could offer different toggeable options such as:
✅ exact duplicates
✅ same extension, similar size
✅ same extension, similar name and size
but I don't know what would be a good default tolerance

**Comment by Worgle123** [comment]
> It could be possible to use a hash value to check. but that would only find exact duplicates in file data.

Could you explain this please? Excuse my ignorance, but what are hash values?

**Comment by Worgle123** [comment]
> It could offer different togglable options such as: ✅ exact duplicates ✅ same extension, similar size ✅ same extension, similar name and size but I don't know what would be a good default tolerance

I agree with you. As for tolerance, maybe it could have different sensitivities for separate extensions? .docx .txt and other such files would need more accurate scans, as their size is generally pretty similar, but with images, sizes tend to vary more. I believe that I already suggested something similar in the original suggestion.

**Comment by Jay-o-Way** [comment]
For those who need it: CCleaner has this function. And i bet a number of others will have it too.

**Comment by yaira2** [comment]
It might be useful to notify the user of duplicates in the downloads folder.

**Comment by tauheedul** [comment]
Thanks for attaching my request, I didn't see the older entry. 

In summary my suggestion was a right click option on the context menu which lists any duplicates in the folder and its nested directories. Displaying any repeated files in the result collection.
